blueprints/incredible-squaring-eigenlayer/src/tests.rs

```diff
@@ -0,0 +1,344 @@
+        *owner_address,
+        *aggregator_address,
+        *task_generator_address,
+    )
+    .await;
+
+    let num_successful_responses_required = 3;
+    let successful_responses = Arc::new(Mutex::new(0));
+    let successful_responses_clone = successful_responses.clone();
+
+    // Start the Task Response Listener
+    let response_listener = setup_task_response_listener(
+        task_manager_address,
+        ws_endpoint.clone(),
+        successful_responses,
+    )
+    .await;
+
+    // Start the Task Spawner
+    let task_spawner = setup_task_spawner(
+        task_manager_address,
+        registry_coordinator_address,
+        *task_generator_address,
+        accounts.to_vec(),
+        http_endpoint.clone(),
+    )
+    .await;
+
+    tokio::spawn(async move {
+        task_spawner.await;
+    });
+
+    tokio::spawn(async move {
+        response_listener.await;
+    });
+
+    info!("Starting Blueprint Binary...");
+
+    let blueprint_process_manager = BlueprintProcessManager::new();
+    let current_dir = std::env::current_dir().unwrap();
+    let xsquare_task_program_path = PathBuf::from(format!(
+        "{}/../../target/release/incredible-squaring-blueprint-eigenlayer",
+        current_dir.display()
+    ))
+    .canonicalize()
+    .unwrap();
+
+    let tmp_dir = tempfile::TempDir::new().unwrap();
+    let keystore_path = &format!("{}", tmp_dir.path().display());
+
+    blueprint_process_manager
+        .start_blueprints(
+            vec![xsquare_task_program_path],
+            &http_endpoint,
+            ws_endpoint.as_ref(),
+            Protocol::Eigenlayer,
+            keystore_path,
+        )
+        .await
+        .unwrap();
+
+    // Wait for the process to complete or timeout
+    let timeout_duration = Duration::from_secs(300);
+    let result = wait_for_responses(
+        successful_responses_clone.clone(),
+        num_successful_responses_required,
+        timeout_duration,
+    )
+    .await;
+
+    // Check the result
+    if let Ok(Ok(())) = result {
+        info!("Test completed successfully with {num_successful_responses_required} tasks responded to.");
+        blueprint_process_manager
+            .kill_all()
+            .await
+            .unwrap_or_else(|e| {
+                error!("Failed to kill all blueprint processes: {:?}", e);
+            });
+    } else {
+        panic!(
+            "Test timed out after {} seconds with {} successful responses out of {} required",
+            timeout_duration.as_secs(),
+            successful_responses_clone.lock().await,
+            num_successful_responses_required
+        );
+    }
+}
+
+/// Sets up the test environment for the EigenLayer Blueprint.
+///
+/// # Description
+/// - Sets all the necessary environment variables for the necessary EigenLayer Contract Addresses.
+/// - Returns a [`EigenlayerTestEnvironment`] struct containing the test environment state.
+pub async fn setup_eigenlayer_test_environment(
```

Comment:
Why was this function specifically moved out of test utils? Is it not something to be reused across all eigen blueprint tests?